### DISCUSSION TITLE
New Method to Solarposition

### DISCUSSION THREAD

**Discussion by NadineFRB** [pvlib/pvlib-python#1714] [Q&A]
Hello All,

I am trying to add one of Grena's solar position algorithm  to the methods of solarposition in PVLib.  I found a very useful information in https://github.com/david-salac/Fast-SZA-and-SAA-computation/blob/master/sza_saa_grena.py, However, I'm having trouble to add it. I wonder if one of the group have already done this kind of simulation and if you could help me.

**Comment by mikofski** [comment]
The license send you can copy the existing code as long as you include the license?
https://github.com/david-salac/Fast-SZA-and-SAA-computation/blob/master/LICENSE

The reamer has a reference we can use according to pvlib traditions: https://www.sciencedirect.com/science/article/abs/pii/S0038092X12000400?via%3Dihub

I am okay with this, any others have concerns? I think it would be nice to also add a benchmark and gallery example perhaps comparing to SPA or SOLPOS

**Comment by mikofski** [comment]
this is related  to #1308
